api-server/web-server/src/api/json_helpers.rs

```diff
@@ -148,3 +153,22 @@ pub fn tx_to_json(tx: &Transaction, chain_config: &ChainConfig) -> serde_json::V
     .iter().map(|out| txoutput_to_json(out, chain_config)).collect::<Vec<_>>()
     })
 }
+
+pub fn block_header_to_json(block: &Block) -> Result<serde_json::Value, ApiServerWebServerError> {
+    let target = match block.header().header().consensus_data() {
+        ConsensusData::PoS(pos) => pos.compact_target(),
+        ConsensusData::PoW(_) | ConsensusData::None => {
+            return Err(ApiServerWebServerError::ServerError(
+                ApiServerWebServerServerError::InternalServerError,
+            ));
+        }
```

Comment:
This effectively disables the whole `block` call for `PoW` and no consensus. Would it be possible to return a dummy value (json `null`) or just omit the target field completely in that case?

Reply:
Or write their json values... why not?